### PR TITLE
Tts added and working

### PR DESCRIPTION
**Pull request by Nikitak211**

### CODE REVIEW

File: src/Homepage.js [hunk 1]
```diff
@@ -1,11 +1,29 @@
 import { useState, useEffect } from "react";
 
+let readSpeech = false;
+let speech = new SpeechSynthesisUtterance();
+speech.lang = "en";
```
Comment (8o8inCodes) [inline src/Homepage.js]: Not good, you need to have these fields inside the component, not here.
useState for the readSpeech boolean
and useRef for the SpeechSynthesisUtterance, and language, you need to make sure you initialize these in useEffect that happens once

File: src/Homepage.js [hunk 2]
```diff
@@ -27,6 +45,8 @@ const Homepage = () => {
                 if (data.success) {
                     setText(data.message)
                     setFailed("")
+                    speech.text = data.message;
+                    if (readSpeech) window.speechSynthesis.speak(speech)
                 }
                 if (data.failure) {
                     setFailed(data.error)
```
Comment (8o8inCodes) [inline src/Homepage.js:58]: Wrong indentation

File: src/Homepage.js [hunk 3]
```diff
@@ -27,6 +45,8 @@ const Homepage = () => {
                 if (data.success) {
                     setText(data.message)
                     setFailed("")
+                    speech.text = data.message;
+                    if (readSpeech) window.speechSynthesis.speak(speech)
```
Comment (8o8inCodes) [inline src/Homepage.js]: Change the name of the boolean to "muted" instead, more informative.

**Review by 8o8inCodes — CHANGES_REQUESTED** [review]

File: src/Homepage.js [hunk 4]
```diff
@@ -1,18 +1,43 @@
...
 const Homepage = () => {
     const [amountOfSnippets, setamountOfSnippets] = useState('')
     const [text, setText] = useState('')
     const [failed, setFailed] = useState('')
-    const url = "/api/generateText";
+    const [soundText, setSoundText] = useState('')
+    const [button, setButton] = useState('')
+    const [muted, setMuted] = useState(false)
+
+    const url = useRef("/api/generateText");
```
Comment (8o8inCodes) [inline src/Homepage.js]: No need to useRef for url, you can just hard code it in the fetch it's self, this line is useless

File: src/Homepage.js [hunk 5]
```diff
@@ -1,18 +1,43 @@
...
+    }
 
     const handleSubmit = async () => {
 
         let headers = new Headers();
 
         headers.append('Content-Type', 'application/json');
 
-        await fetch(url, {
+        await fetch(url.current, {
```
Comment (8o8inCodes) [inline src/Homepage.js]: Just write a url here instead of url.current

File: src/Homepage.js [hunk 6]
```diff
@@ -42,7 +68,11 @@ const Homepage = () => {
     }
 
     useEffect(() => {
-        console.log('Entry was made')
+        setButton("On");
+        setSoundText("speech Off")
+        if (typeof window !== 'object' || !window.speechSynthesis) return;
```
Comment (8o8inCodes) [inline src/Homepage.js]: What is this for?

**Review by 8o8inCodes — CHANGES_REQUESTED** [review]

File: src/Homepage.js (hunk 6 above)
Comment (Nikitak211) [reply src/Homepage.js]: So on load it will display The "On" and "speech Off"

**Review by 8o8inCodes — APPROVED** [review]

**Pull request merged**